### ISSUE TITLE
No other packages should be installed to base

### ISSUE THREAD

**Issue by abalter** [mamba-org/mamba#2208]
### Troubleshooting docs

- [X] My problem is not solved in the Troubleshooting docs

### Search tried in issue tracker

only in base environment

### Latest version of Mamba

- [X] My problem is not solved with the latest version

### Tried in Conda?

Not applicable

### Describe your issue

Cross-posted here: https://superuser.com/questions/1761379/mamba-no-other-packages-should-be-installed-to-base?noredirect=1#comment2738704_1761379

The [mamba documentation](https://mamba.readthedocs.io/en/latest/user_guide/troubleshooting.html#mamba-should-be-installed-to-the-base-environment) says:

> ## No other packages should be installed to base
> Installing packages other than Conda and Mamba into the base environment is not supported. Mamba must live in the same environment as Conda, and Conda does not support having packages other than Conda itself and its dependencies in base.

I want to make sure I'm understanding this correctly, as this is different from how I've always set up conda.

Does this mean that from the time I initially install mamba I should only use the base environment for creating other environments? Would I then want to designate some other environment, maybe call it "default," as the one that loads through my linux profile?

## Note
Tags for "conda" and "mamba" don't exist yet. Although this is a sysadmin type question, not programming, I'm wondering if I should post instead to stackoverflow?

### mamba info / micromamba info

```shell
N/A
```

### Logs

```shell
N/A
```

### environment.yml

```yaml
N/A
```

### ~/.condarc

```yaml
N/A
```

**Comment by jonashaag** [comment]
I‘m not sure I understand the question. 

Conda „lives“ in the base environment: Conda itself is a Python program that needs dependencies. Those are installed in the base environment. If you add other things to the base environment than just Conda and its dependencies, you risk breaking your Conda installation. Thus it is strongly recommended not to install any other packages to base.

What does it mean in your terms to „load“ a profile?

**Comment by abalter** [comment]
If the best practice is to ONLY use the base environment to use other environments, I honestly thing no one knows that. In fact, I always thought that is why everyone I know has the base environment loaded by default in their .bashrc.

Being able to make specific virtual environments is great. I use that feature. But conda developed into a full-on package manager. As a data scientist, and especially in the bio field, I use software written in a variety of languages, and released with different levels of sophistication. One option would be to install some with apt, some from binaries, and some build from source after figuring out all of the dependencies. Then pray that somehow all of these packages work together. I have seen bioinformaticists tear their hair out doing this. I just install everything with conda.

If I understand you correctly, the way we are _supposed_ to use conda or mamba is to NEVER use the base environment other than to create new environments. I have never heard that before and never seen anyone do that. If that is the case, what do you suggest we use as our _default_ environment? I need to have _some_ environment loaded when I start my shell so I can run programs interactively in a casual way. 

Should I create an environment, say `mybase` to be my default which loads when I start the shell, and then switch to _base_ if I want to create a new environment? If I want to install a package into `mybase` while I'm working in that environment, can I run `mamba install xxxx -y`, or do I switch to _base_ first, and then run `mamba install -n mybase xxxx -y`?

**Comment by abalter** [comment]
Well, I just looked at the official [conda docs](https://conda.io/projects/conda/en/latest/user-guide/getting-started.html#managing-environments), and they explicitly say: "You don't want to put programs into your base environment, though."

So, I've definitely been using conda in a way that it was not intended.

So, here is a word for the conda/mamba community. I mostly have experience working in the bioinformatics/biostatistics world. Using conda/mamba for controlled and reproducible environments is crucial for fairly static pipelines and specific repeatable tasks. However, most of our time is spent just "working" of means lots of exploratory stuff running bits of this and bits of that. Trying out new packages. Testing new methods. A huge proportion of bioinformatics papers and biostatistics papers will be a new methods or software implementation. Because creating a new environment for every type of programming or research "mood" we may be in that day is ridiculous, we spend a lot of time in our "base" environments.

Here is my suggestion:

1. Make it clearer that the _base_ environment should only be used for creating new environments (and I'm still not exactly sure what that means as I described above).
2. No longer prompt to have the _base_ environment loaded by default. That will encourage people to use it as their _default_ environment. I know that because that is what most people do.
3. Consider developing and publishing some best-practices around the use case I described above which, I think, probably describes between 60% an 90% of conda/mamba usage.
4. Reformulate the entire system so that you naturally have a default environment which you are prompted to initialize in your .bashrc (or however it works in Windows). However, when you type a command to create an environment or install/update/remove packages the machinery of doing `mamba deactivate; mamba activate base; mamba create -n newenv r=4.2.2` is automatically done when you run `mamba create -n newenv r=4.2.2` so the "best practice" is handles by the package manager itself.

What do you think?

**Comment by jonashaag** [comment]
I agree! But this is nothing we can change in Mamba. It’s a Conda problem. I consider it a design flaw in Conda. 

If you use Micromamba you don’t have a base environment that you need to „keep clean“ because Micromamba lives outside of any environment. I personally have a _Micromamba_ environment that I call „base“ that’s what you call „mybase“.

For Conda/Mamba, I think the „mybase“ approach is common.

Btw you don’t need to switch to base (or even activate any environment whatsoever) to create new environments or to activate other environments. You can run (almost?) all Conda/Mamba commands without having any environment activated by passing `-n`.

**Comment by abalter** [comment]
@jonashaag -- thanks for the clarification and micromamba suggestion. I misunderstood how rigorously one needed to use the base environment. So it has to stay clean, but you can do all operations in any environment, as long as you don't pollute the base.

**Comment by jonashaag** [comment]
Yep!

**Comment by jcpayne** [comment]
@abalter is right; this needs a far more comprehensive discussion in the documentation.  I was also confused by the offhand remark, and still have questions; for example, if conda and mamba both have their own dependencies, how can a user tell if they installed a package in *base* by mistake, or whether the packages in *base* properly belong to either conda or mamba?

**Comment by jonashaag** [comment]
My recommendation is to use Micromamba or Pixi.

**Comment by jcpayne** [comment]
Thank you.  Is there a micromamba equivalent for a [pip editable install](https://stackoverflow.com/questions/278081/resolving-a-git-conflict-with-binary-files)?  Or failing that, some way to make micromamba pay attention to the packages that pip installs?  I'd like to be able to clone a code repository that changes frequently, and the authors recommend using an editable install for that.

**Comment by jonashaag** [comment]
No and yes, Conda/Mamba should consider things installed with pip

**Comment by jtskowronek** [comment]
@jonashaag My search has found this issue thread as a best fit, but can you direct me to another thread regarding mamba installing packages into base from a non-base env which is currently activated in my shell? I finally realized this has been happening at random, possibly the source of many of my mamba headaches. Thanks.
